app/src/main/java/com/example/myapplication/mvp/presenter/CalculatorPresenter.kt

```diff
@@ -5,55 +5,106 @@ import com.example.myapplication.mvp.view.CalculatorView
-                model.setOperatorTwo(EMPTY_STRING)
-                model.setOperand(operator)
-                view.setVisor("${model.getOperatorOne()}$operator")
+        if (model.result == ZERO_FLOAT_RESULT) {
+            if (model.operand.isNotEmpty()) {
+                model.operatorTwo = EMPTY_STRING
             }
+            model.operand = operator
+            view.setVisor("${model.operatorOne}$operator")
+            view.activePoint()
+        } else if (model.operand.isEmpty()) {
+            model.operand = operator
+            model.operatorOne = model.result.toString()
+            model.result = ZERO_FLOAT_RESULT
+            view.setVisor("${model.operatorOne}$operator")
+            view.setResult(EMPTY_STRING)
         }
     }
 
     fun onEqualPressed() {
-        when (model.getOperand()) {
-            PLUS -> {
-                model.setResult(model.getOperatorOne().toFloat() + model.getOperatorTwo().toFloat())
-                view.setResult(model.getResult().toString())
-            }
-            SUBTRACT -> {
-                model.setResult(model.getOperatorOne().toFloat() - model.getOperatorTwo().toFloat())
-                view.setResult(model.getResult().toString())
-            }
-            DIVIDE -> {
-                model.setResult(model.getOperatorOne().toFloat() / model.getOperatorTwo().toFloat())
-                view.setResult(model.getResult().toString())
-            }
-            MULTIPLY -> {
-                model.setResult(model.getOperatorOne().toFloat() * model.getOperatorTwo().toFloat())
-                view.setResult(model.getResult().toString())
+        if (model.operatorOne.isNotEmpty() && model.operatorTwo.isNotEmpty()) {
+            if (model.operatorOne != POINT && model.operatorTwo != POINT) {
+                when (model.operand) {
+                    PLUS -> {
+                        model.result = model.operatorOne.toFloat() + model.operatorTwo.toFloat()
+                        view.setResult(model.result.toString())
+                    }
+                    SUBTRACT -> {
+                        model.result = model.operatorOne.toFloat() - model.operatorTwo.toFloat()
+                        view.setResult(model.result.toString())
+                    }
+                    DIVIDE -> {
+                        if (model.operatorTwo.toFloat() != ZERO_FLOAT_RESULT) {
+                            model.result = model.operatorOne.toFloat() / model.operatorTwo.toFloat()
+                            view.setResult(model.result.toString())
+                        } else {
+                            view.showToastError(TOAST_ZERO_DIVIDE)
+                        }
+                    }
+                    MULTIPLY -> {
+                        model.result = model.operatorOne.toFloat() * model.operatorTwo.toFloat()
+                        view.setResult(model.result.toString())
+                    }
+                }
+                model.operatorOne = EMPTY_STRING
```

Comment:
this could be a public method such as `reset()` instead of externally modifying the internal state of the model

Reply:
yes! it´s true, done !